Add option to move away from endstop after homing.

marcio-ao opened the pull request:
- Added ENDSTOP_BACKOFF_X, ENDSTOP_BACKOFF_Y and ENDSTOP_BACKOFF_Z
- This can help with printers that have a recessed Z home button (Lulzbot TAZ 6)
- This can also avoid chattering that happens when a printer with sensorless
  homing is re-homed when the axis are already sitting against the bumpers.

This is a more general feature than the one requested in #11748 and is a cleaned up implementation based on the FW for our TAZ 6 and Mini printers.

marcio-ao committed:
Add option to move away from endstop after homing.

- Added ENDSTOP_BACKOFF_X, ENDSTOP_BACKOFF_Y and ENDSTOP_BACKOFF_Z
- This can help with printers that have a recessed Z home button (Lulzbot TAZ 6)
- This can also avoid chattering that happens when a printer with sensorless
  homing is re-homed when the axis are already sitting against the bumpers.

robbycandra commented:
one of my product using 1284P is also move away from endstop after homing but using different method. My goal is to combine Limit X PIN and Y PIN and I can use the spare pin for filament detector.

thinkyhead commented:
This might be better covered by adding `EVENT_GCODE_POST_HOME` so users can not just back off, but move to a specific position, or park, or do a tool-change, or change the lighting, etc….

thinkyhead committed:
Fix and simplify ENDSTOP_BACKOFF

- Reduce to a single option
- Supplant broken ANY() test
- Only `endstops.enable(false)` (followed by `not_homing`) needed
- Set default to back off 2mm from all endstops (disabled)

thinkyhead commented:
Hmm, well, if the back-off is meant to occur as part of `G28` then this feature does make that very simple. The Events-G-code stuff will come together later.

I've reduced the feature to a single array option and the code is simplified. Please test to make sure it still behaves the same. There was no feedrate option in the config so it just uses the default `do_blocking_move_*` rates, which should be reasonable. If a feedrate option is needed, that may be added.

AnHardt commented:
What would this mean for kinematic systems?
Seems to be similar to the DELTA endstop offsets.
Should the software endstops be set at the new position?
Would shifting to `homeaxis()` save some code?
